templates/Dockerfile.j2

```diff
@@ -9,13 +9,12 @@ RUN yum update -y && yum install -y curl && yum clean all
 
 RUN curl -Lso - {{ url }} | \
       tar zxf - -C /tmp && \
-      mv /tmp/apm-server-{{ version }}-linux-x86_64 {{ home }}
+      mv /tmp/apm-server-{{ version }}-linux-x86_64 {{ home }} && \
+      echo 'output.elasticsearch.hosts: ["elasticsearch:9200"]' >> {{ home }}/apm-server.yml
```

Comment:
This appends the `output.elasticsearch.hosts` entry to the `apm-server.yml` file.

Reply:
Doesn't this leave two `output.elasticsearch.hosts` declarations in the file? A user could easily be distracted/confused by one of them. I think this particularly bad, since the original one has a bunch of comments around it to guide the user:

```
#================================ Outputs =====================================

# Configure what output to use when sending the data collected by the beat.

#-------------------------- Elasticsearch output ------------------------------
output.elasticsearch:
  # Array of hosts to connect to.
  hosts: ["localhost:9200"]
```
The fact that the last configuration wins will not help a user that is following our instructions here in the file. I beleive we should always consider the computer the be the secondary audience, and optimize for the reader/user.

@simitt's `sed` approach gives a cleaner result. What was the motivation to not do that?

Reply:
If you are worried about the specificity of the original regexp (fair enough), you could use a multiline approach, like this:

```
sed -zr 's/output.elasticsearch:(\n[^\n]*){2}/output.elasticsearch:\n  hosts: ["elasticsearch:9200"]/'
```

That version also preserves the original, nested format, which is a bonus.'\

Here's a more robust version, which is admittedly getting hard to read. It will, however, cope if the comments or whitespace change above the `hosts:` line.

```
sed -zr 's/output.elasticsearch:\n(\n| +\n| +#[^\n]*)* +hosts[^\n]*/output.elasticsearch:\n  hosts: ["elasticsearch:9200"]/'
```

Probably overkill.

Reply:
Sorry, @graphaelli, I sent my second message before I saw yours.

Failing silently is definitely a possibility if we are not careful, and definitely bad. We have pretty good test coverage for the images to guard against these sorts of things.

I proposed this test for the config manipulation: https://github.com/simitt/apm-server-docker/pull/1/files . I think we should have that test regardless of how we implement the config change.

Reply:
Also, I reckon that a user who bind-mounts the config will want to start with an example. I, personally, would tend to copy the config out of the image as my starting point.

Reply:
Whoa, I was not aware we could add tests on this, that's awesome!  The [first multline sed](https://github.com/elastic/apm-server-docker/pull/12#discussion_r179310732) + simitt/apm-server-docker#1 looks like a winner to me.

Reply:
😁 Thanks.
I'm a firm believer in the mantra the _you can add tests to anything_. It's fun game figuring out how.